x-pack/plugins/security_solution/cypress/tasks/create_new_rule.ts

```diff
@@ -210,48 +237,14 @@ export const fillAboutRuleWithOverrideAndContinue = (rule: OverrideRule) => {
 
   cy.get(DEFAULT_RISK_SCORE_INPUT).type(`{selectall}${rule.riskScore}`, { force: true });
 
-  rule.tags.forEach((tag) => {
-    cy.get(TAGS_INPUT).type(`${tag}{enter}`, { force: true });
-  });
+  if (rule.tags) fillRuleTags(rule.tags);
 
   cy.get(ADVANCED_SETTINGS_BTN).click({ force: true });
 
-  rule.referenceUrls.forEach((url, index) => {
-    cy.get(REFERENCE_URLS_INPUT).eq(index).type(url, { force: true });
-    cy.get(ADD_REFERENCE_URL_BTN).click({ force: true });
-  });
-
-  rule.falsePositivesExamples.forEach((falsePositive, index) => {
-    cy.get(FALSE_POSITIVES_INPUT).eq(index).type(falsePositive, { force: true });
-    cy.get(ADD_FALSE_POSITIVE_BTN).click({ force: true });
-  });
-
-  let techniqueIndex = 0;
-  let subtechniqueInputIndex = 0;
-  rule.mitre.forEach((mitre, tacticIndex) => {
-    cy.get(MITRE_ATTACK_TACTIC_DROPDOWN).eq(tacticIndex).click({ force: true });
-    cy.contains(MITRE_TACTIC, mitre.tactic).click();
-
-    mitre.techniques.forEach((technique) => {
-      cy.get(MITRE_ATTACK_ADD_TECHNIQUE_BUTTON).eq(tacticIndex).click({ force: true });
-      cy.get(MITRE_ATTACK_TECHNIQUE_DROPDOWN).eq(techniqueIndex).click({ force: true });
-      cy.contains(MITRE_TACTIC, technique.name).click();
-
-      technique.subtechniques.forEach((subtechnique) => {
-        cy.get(MITRE_ATTACK_ADD_SUBTECHNIQUE_BUTTON).eq(techniqueIndex).click({ force: true });
-        cy.get(MITRE_ATTACK_SUBTECHNIQUE_DROPDOWN)
-          .eq(subtechniqueInputIndex)
-          .click({ force: true });
-        cy.contains(MITRE_TACTIC, subtechnique).click();
-        subtechniqueInputIndex++;
-      });
-      techniqueIndex++;
-    });
-
-    cy.get(MITRE_ATTACK_ADD_TACTIC_BUTTON).click({ force: true });
-  });
-
-  cy.get(INVESTIGATION_NOTES_TEXTAREA).type(rule.note, { force: true });
+  if (rule.referenceUrls) fillReferenceUrls(rule.referenceUrls);
+  if (rule.falsePositivesExamples) fillFalsePositiveExamples(rule.falsePositivesExamples);
+  if (rule.mitre) fillMitre(rule.mitre);
+  if (rule.note) fillNote(rule.note);
```

Comment:
Nit: it would be ideal to do general refactorings in separate PRs to reduce the cognitive load. But thank you for making it cleaner, I really appreciate it!

Also nit: for JS/TS codebases (including Kibana) this is a more typical code style:

```ts
if (rule.referenceUrls) {
  fillReferenceUrls(rule.referenceUrls);
}
```

Reply:
You are right regarding doing refactors in a different PR, due to my limited bandwidth and how long a PR takes to be merged sometimes, I decided to go ahead and do everything on this one.